research/datasets.py

```diff
@@ -1,19 +1,66 @@
 from functools import partial
 from typing import Literal, Optional
+from dataclasses import dataclass
+import copy
 
 import torch
 from torch.utils.data import DataLoader
 
 from lizrd.text import datasets, packers, data, tokenizers
+from lizrd.support.misc import get_ith_chunk
+
+
+@dataclass
+class BatchSizeRampupConfig:
```

Comment:
My initial suggestion was to replace two list with a list of tuples. 
For  transition_points = [0.5, 1, 2]  and batch_sizes = [16, 64, 128] representing
```math
f(processedtokens) = 
     \begin{cases}
       \text{16} &\quad\text{if} 0 \leq \text{processedtokens} \le 0.5 \\
       \text{64} &\quad\text{if} 0.5 \leq \text{processedtokens} \le 1 \\
       \text{128} &\quad\text{if} 1 \leq \text{processedtokens} \le 2 \\
       \text{512} &\quad \text{otherwise}
     \end{cases}
```

Sorry if I wasn't clear.

Reply:
I may have misunderstood, sorry. I think we can easily rethink/improve this part in the future